src/module/apps/characterImport/vehicleImport/VehicleParser.ts

```diff
@@ -2,31 +2,34 @@ import { getArray } from "../importHelper/BaseParserFunctions.js";
 import { WeaponParser } from "../weaponImport/WeaponParser";
 import { GearsParser } from "../importHelper/GearsParser.js";
 import { MountedWeaponParser } from "./MountedWeaponParser.js";
+import { SR5Actor } from '../../../actor/SR5Actor';
 
 export default class VehicleParser {
 
-    async parseVehicles(actor, chummerChar, importOptions) {
-
+    async parseVehicles(actor, chummerChar, importOptions): Promise<Array<SR5Actor>|undefined> {
         if(!importOptions.vehicles) {
             return;
         }
 
-        if(!game.user.can("ACTOR_CREATE")) {
-            ui.notifications.error(game.i18n.format("SR5.VehicleImport.MissingPermission"))
+        if(!game.user?.can("ACTOR_CREATE")) {
+            ui.notifications?.error(game.i18n.format("SR5.VehicleImport.MissingPermission"))
             return;
         }
 
         const vehicles = getArray(chummerChar.vehicles?.vehicle);
 
-        vehicles.forEach(async vehicle => {
-            let vehicleActor = await Actor.create({
+        return await Promise.all<SR5Actor>(vehicles.map<Promise<SR5Actor>>(async (vehicle) => {
```

Comment:
awaiting here does slow the import process and gives no benefit in my view

Reply:
It totally makes sense to not perform all of these operations serially, but wouldn't we still hope to have all operations done by the time `CharacterImporter.importChummerCharacter` returns? It seems to me like we're opening ourselves up to potential issues due to race conditions by not carefully managing our promises (although I concede that the character importer is unlikely to have any problems in that regard).

Admittedly I mostly made that change for ease of testing, so that I could get a list of all of the actors created by the parser.  I don't feel strongly and I can change it back if you'd like.  That all being said, since we don't currently await on the result of `new VehicleParser().parseVehicles` in `CharacterImporter` anyway, I don't think any performance impact is incurred by this change.

Reply:
I think this is fine then. Huge amount of vehicles should be expected to take a while anyway :D

Reply:
Thanks for moving it to TypeScript as well ;D